src/qibolab/instruments/rfsoc/driver.py

```diff
@@ -386,14 +400,12 @@ def merge_sweep_results(
         """
         for serial in dict_b:
             if serial in dict_a:
-                cls = dict_a[serial].__class__
-                if isinstance(dict_a[serial], IntegratedResults):
-                    new_data = np.column_stack(
-                        [dict_a[serial].voltage, dict_b[serial].voltage]
-                    )
-                elif isinstance(dict_a[serial], SampleResults):
-                    new_data = np.append(dict_a[serial].samples, dict_b[serial].samples)
-                dict_a[serial] = cls(new_data)
+                data = lambda res: (
+                    res.voltage if isinstance(res, IntegratedResults) else res.samples
+                )
+                dict_a[serial] = type(dict_a[serial])(
+                    np.append(data(dict_a[serial]), data(dict_b[serial]))
+                )
             else:
                 dict_a[serial] = dict_b[serial]
         return dict_a
```

Comment:
Not this PR business, but all this function would be just an `np.append` (or `np.stack`, along the out-enough dimension) if the results' objects were simple NumPy arrays...

(possibly even just an `np.append`, and later reshape/transpose...)